sdk-setup/src/mb2

```diff
@@ -2739,11 +2776,58 @@ run_package() {
 }
 
 run_build_shell() {
+    local maintain=
+
+    while (($# > 0)); do
+        case $1 in
+            --maintain)
+                maintain=1
+                ;;
+            --)
+                break
+                ;;
+            -*)
+                fatal "Unrecognized option: '$1'"
+                ;;
+            *)
+                break
+                ;;
+        esac
+        shift
+    done
+
+    if [[ $maintain ]]; then
+        sb2_maintain "$@"
+    else
+        sb2_build "$@"
+    fi
+}
+
+sb2_build() {
+    local sb2_args=()
+    if [[ $OPT_BUILD_SHELL_ARGS ]]; then
+        read -a sb2_args <<<$OPT_BUILD_SHELL_ARGS || return
+    fi
+
     if [[ $# -gt 0 ]]; then
-        oomadvice sb2 -t $OPT_TARGET -- "$@"
+        oomadvice sb2 -t $OPT_TARGET "${sb2_args[@]}" -- "$@"
     else
         # Keep PS1 aligned with `sdk-manage maintain`
-        sb2 -t $OPT_TARGET -- env PS1="[$OPT_ORIGINAL_TARGET] \W $ " /bin/bash --noprofile --norc
+        sb2 -t $OPT_TARGET "${sb2_args[@]}" -- env PS1="[$OPT_ORIGINAL_TARGET] \W $ " /bin/bash --noprofile --norc
+    fi
+}
+
+sb2_maintain() {
+    local sb2_args=(-m sdk-install -R)
+    if [[ $OPT_BUILD_SHELL_MAINTENANCE_ARGS ]]; then
+        read -a sb2_args <<<$OPT_BUILD_SHELL_MAINTENANCE_ARGS || return
+    fi
+
+    if [[ $# -gt 0 ]]; then
+        sb2 -t $OPT_TARGET "${sb2_args[@]}" -- "$@"
```

Comment:
Why is "-t $OPT_TARGET" not part of sb2_args? It seems a bit random to handle it separately.

Reply:
For some reason I somehow thought `read -a sb2_args <<<$OPT_BUILD_SHELL_MAINTENANCE_ARGS` would append to sb2_args, instead of replacing existing values. I blame lack of coffee.